### DISCUSSION TITLE
How to add custom headers to /_next/static/chunks/***.js?

### DISCUSSION THREAD

**Discussion by weyert** [vercel/next.js#19146] [Help]
I am wondering how I can add custom headers such as HSTS and X-Frame-Options to the `/_next/static/chunks/***.js` files? I tried to use something like what is listed below in `next.config.js` but it is not working. 

What am I doing wrong?

```javascript
module.exports = withBundleAnalyzer({
  distDir: 'build',

  // Add security headers to the web-application
  headers() {
    return Promise.resolve([
      {
        source: '/',
        headers: createSecureHeaders({
          // allow embedding the iframe widget
          frameGuard: false,
        }),
      },
      {
        source: '/_next(.*)',
        headers: createSecureHeaders(),
      },
    ])
  },

  // Attempt to inject polyfills as part of the main polyfills file that is bundled with the Next.js
  webpack: (config, { dev: isDevelopmentMode }) => {
    const originalEntry = config.entry
    config.entry = async () => {
      const entries = await originalEntry()
      if (isDevelopmentMode) {
        if (entries['main.js'] && !entries['main.js'].includes('./src/polyfills.development.ts')) {
          entries['main.js'].push('./src/polyfills.development.ts')
        }
      }

      return entries
    }

    // Attempt to ignore storybook files when doing a production build,
    // see also: https://github.com/vercel/next.js/issues/1914
    if (!isDevelopmentMode) {
      config.module.rules.push({
        test: /\.stories.(js|tsx?)/,
        loader: 'ignore-loader',
      })
    }

    return config
  },
})
```

**Comment by federicom-scale** [comment]
Same question here, I want to set cache-control on /_next/static/chunks/* since its not caching the bundles for development

**Comment by jrmyio** [comment]
Here in 2022, this still seems to be an issue. It's refreshing an SVG used in a background-image of a styled component.  Normally you can get around this by adding the headers:

```json
"headers": [
        {
            "source": "/:all*(.woff2|.woff|.ttf|.eot|.svg|.png|.jpg|.gif|.webp)",
            "headers": [
                {
                    "key": "Cache-Control",
                    "value": "public, max-age=31536000, immutable"
                }
            ]
        }
    ]
```

But this doesn't seem to work on `/_next/static/chunks/*`.

**Comment by bharadwajpro** [comment]
Anyone found a solution? I am trying to cache few "wasm" files by adding the custom header. But it doesn't work.

**Comment by mmaldonadodev** [comment]
I Have the same problem without any solution.

**Reply by Bracciata** [reply]
Also having the same problem in 2024!

**Comment by Bracciata** [comment]
Is this possibly related to https://github.com/vercel/next.js/issues/55129 and fixed by upgrading Next Version?

**Comment by dcimorra** [comment]
There is the option to do this through apache/nginx, but ideally next should not be closed to allow modification of the chunk headers.

![image](https://github.com/user-attachments/assets/59c410b6-0172-4101-9988-403925a040da)
